californium-core/src/main/java/org/eclipse/californium/core/CoapObserveRelation.java

```diff
@@ -125,8 +126,9 @@ public boolean reregister() {
 		}
 		if (registrationPending.compareAndSet(false, true)) {
 			Request refresh = Request.newGet();
-			refresh.setDestination(request.getDestination());
-			refresh.setDestinationPort(request.getDestinationPort());
+			EndpointContext destinationContext = response != null ? response.advanced().getSourceContext()
+					: request.getDestinationContext();
+			refresh.setDestinationContext(destinationContext);
```

Comment:
Why not always using request ?

Reply:
Because the EndpointContext of the request is just the initial one. Usually the "completed" EndpointContext is passed back to the Exchange. FMPOV further request in the "context of the observe" should use that complete EndpointContext, if available.

Reply:
ok.